firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigFetchHandler.java

```diff
@@ -530,11 +538,14 @@ private Long getFirstOpenTime() {
     return (Long) connector.getUserProperties(/*includeInternal=*/ true).get(FIRST_OPEN_TIME_KEY);
   }
 
+  // Used to initialize templateVersion.
+  private long getTemplateVersionFromDisk() {
+    // Get templateVersion on disk from metadata.
+    return frcMetadata.getLastTemplateVersion();
+  }
+
   public long getTemplateVersionNumber() {
-    if (fetchedConfigsCache.get() != null && fetchedConfigsCache.get().getResult() != null) {
-      return fetchedConfigsCache.get().getResult().getTemplateVersionNumber();
-    }
-    return 1L;
+    return lastTemplateVersion;
```

Comment:
If the goal is to provide the last fetched template version, I would encapsulate it in `FetchResponse` with the other fetched data like `lastFetchETag`

Reply:
None of the fetch response itself isn't kept around after it's invoked. And lastFetchEtag is invoked once to store on disk. I keep the templateVersion as an instance var so that it can be accessed quickly and frequently from Realtime.

Reply:
Gotcha. We could follow the same pattern as we do with the ETag and read it from metadata when it's needed (even though it might read from disk). That way there's just one source of truth. wdyt?

https://github.com/firebase/firebase-android-sdk/pull/4051/files#diff-d171ba9c7617c7b7a8ac69a1b277477a505e618633831ba0e9d828e609b87097R318

Reply:
I think that makes sense, I updated it to reflect this